### DISCUSSION TITLE
Implementing typesafety through out the codebase.

### DISCUSSION THREAD

**Discussion by SAMAD101** [borgbase/vorta#1868] [Ideas]
## Typesafety
Using `typing` and implementing type safety in the project can be pretty useful in the long run as it improves readability, and can act as documentation within the code. Also, it will help catch any type-related bugs during runtime, therefore cool for development and debugging. 

---
I had an idea to make the whole codebase typesafe, so I decided to open a discussion first to see others' views.

**Comment by m3nu** [comment]
If you want to do the PR and there are no large open PRs that would get conflicts, I have no objections.

**Reply by SAMAD101** [reply]
Cool, I'll get to work then :)

**Reply by real-yfprojects** [reply]
Beware that this is very difficult to achieve since GUI's are instantiated from `.ui` files. Static type linters cannot be teached to understand such dynamic behaviour. One can work around that by [generating code from the ui files](https://doc.qt.io/qtforpython-6/tutorials/basictutorial/uifiles.html) during development already instead of doing it at runtime. However that brings the problem of keeping the .ui files in sync with the generated code. At least a pre-commit hook would be needed to ensure that.

**Reply by SAMAD101** [reply]
True, using `QUiLoader` would not be a good fit as certain elements and manipulated during the runtime. For simpler parts of GUI we can do that, but that will introduce inconsistencies in the code. So, lets just stick with `uic.loadUiType`.
For now, I shall work on some different parts of the code.
